pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -265,6 +267,13 @@ func (a *logMetricAdapter) Update(ctx context.Context, u *unstructured.Unstructu
 		update.BucketName = a.desired.Spec.LoggingLogBucketRef.External
 	}
 
+	diffs, err := ListAnyDiff(a.actual, update)
+	if err != nil {
+		return fmt.Errorf("error computing diffs: %w", err)
+	}
+	log.Info("updating logMetric", "diffs", diffs)
+
+	//")
 	// DANGER: this is an upsert; it will create the LogMetric if it doesn't exists
```

Comment:
Not related to this PR. But are we calling Update API in every reconciliation?

Reply:
We call Update on every reconciliation, but the controller is expected to skip GCP calls if it can figure it out.  That's actually pretty tricky for LoggingLogMetric because of some mutable-but-unreadable fields, so I proposed #1772 which is a cute trick if we can apply it more generally